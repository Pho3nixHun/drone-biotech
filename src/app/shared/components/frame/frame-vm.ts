export interface FrameVM {
  title: string;
}
