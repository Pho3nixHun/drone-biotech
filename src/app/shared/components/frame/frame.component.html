@if (vm(); as vm) {
  <section *transloco="let t" class="flex flex-col gap-y-10 py-20 mx-5 sm:mx-20 lg:mx-40">
    <h2>{{ t(vm.titleKey) }}</h2>
    <ng-content />
  </section>
}
