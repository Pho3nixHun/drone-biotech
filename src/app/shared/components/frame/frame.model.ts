export interface FrameVM {
    titleKey: string;
}
