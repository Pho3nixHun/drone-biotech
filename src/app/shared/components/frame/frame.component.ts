import { Component, input } from '@angular/core';
import { FrameVM } from './frame-vm';
import { TranslocoModule } from '@jsverse/transloco';

/**
 * FrameComponent
 *
 * Type: Presentational
 *
 * Scope:
 * - Responsible for structuring a section that contains dynamic content.
 * - It uses content projection to visualize the components.
 * - Manages the overall layout and styling for the section.
 *
 * Out-of-Scope:
 * - Contains no logic related to the behavior or data of the projected components.
 * - Not responsible for fetching or transforming data.
 *
 * Purpose (optional):
 * To provide a flexible and reusable section layout.
 */
@Component({
  selector: 'app-frame',
  standalone: true,
  templateUrl: './frame.component.html',
  imports: [TranslocoModule],
})
export class FrameComponent {
  vm = input<FrameVM | null>(null, { alias: 'vm' });
}
