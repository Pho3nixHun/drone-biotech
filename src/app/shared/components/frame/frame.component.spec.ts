import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FrameComponent } from './frame.component';
import { Component } from '@angular/core';

@Component({
  template: `
    <app-frame>
      <div>{{ text }}</div>
    </app-frame>
  `,
})
class TestHostComponent {
  text = 'Should be projected';
}

describe('FrameComponent', () => {
  let component: TestHostComponent;
  let fixture: ComponentFixture<TestHostComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [FrameComponent],
      declarations: [TestHostComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(TestHostComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.debugElement.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should contain the projected <div> element', () => {
    console.log(compiled);
    const divElement: HTMLElement | null = compiled.querySelector('app-frame section div');
    expect(divElement?.innerText).toBe(component.text);
  });
});
