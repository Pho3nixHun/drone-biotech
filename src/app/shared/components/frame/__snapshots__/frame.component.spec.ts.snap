// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrameComponent should not render the template when there is no VM provided 1`] = `
<div
  id="root1"
  ng-version="18.2.7"
>
  <app-frame>
    <!--container-->
  </app-frame>
</div>
`;

exports[`FrameComponent should render the template when the VM is provided 1`] = `
<div
  id="root2"
  ng-version="18.2.7"
>
  <app-frame
    ng-reflect-vm="[object Object]"
  >
    <section
      class="flex flex-col gap-y-10 py-20 mx-5 sm:mx-20 lg:mx-40"
    >
      <h2>
        Our Products
      </h2>
      <div>
        TestDiv
      </div>
    </section>
    <!--container-->
    <!--container-->
  </app-frame>
</div>
`;
